### ISSUE TITLE
omjournal does not support templating

### ISSUE THREAD

**Issue by bobthemighty** [rsyslog/rsyslog#769]
Hi,

I'm trying to set up rsyslog to handle json-logstash messages from the syslog docker log driver.

I want to ship raw json to redis or kafka for later processing, but also put a human-readable version of the log into the local journal.

I've got a first, hackish approach to a templated omjournal and I'm happy to fix it up and open a PR, but would you prefer:

1)
``` 
template(name="foo" type="list") {
   property(name="some-expr" outname="message")
   property(name="some-expr" outname="severity")
}

action(type="omjournal" template="foo")
```

where structured data are automatically mapped to the appropriate systemd-journal field, or

2)

```
template(name="journal-message" type="string" string="some-expr") 
template(name="journal-severity" type="string" string="some-expr")
action(type="omjournal" 
    message-template="journal-message" 
    severity-template="journal-severity"
)
```

where the user configures a separate template for each item or 

3) Something else entirely?

I think as an end-user, my preference is 1, but it seems less flexible.

Thoughts?

**Comment by bobthemighty** [comment]
Actually, option 1 makes it really simple to add in arbitrary fields, so I'm going to plough on with that.

**Comment by rgerhards** [comment]
implemented via #770
